src/tigergraph.ts

```diff
@@ -0,0 +1,173 @@
+import { InputNode, InputLink } from './types'
+import 'btoa'
+
+export class TigerGraphConnection {
+  host: string;
+  graphname: string;
+  username: string;
+  password: string;
+  token: string;
+ 
+  constructor(host: string, graphname: string, username: string, password: string, token?: string) {
+    this.host = host;
+    this.graphname = graphname;
+    this.username = username;
+    this.password = password;
+    this.token = "";
+  }
+
+  async generateToken() {
+    return fetch(`${this.host}:9000/requesttoken`, {
+        method: 'POST',
+        body: `{"graph": "${this.graphname}"}`,
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': 'Basic '+btoa(`${this.username}:${this.password}`),
+        }
+    }).then(response => {
+        if (!response.ok) {
+            throw new Error(`Error! status: ${response.status}`);
+        }
+    
+        return response.json();
+    }).then(data => {
+        this.token = data.results.token;
+        return this.token;
+    });
+  }
+
+  async getTigerGraphData(vertex_type: Array<string>, edge_type: Array<string>) : Promise<{ nodes: InputNode[]; links: InputLink[]; }> {
```

Comment:
@GenericP3rson What if we use generics here? Something like
```ts
 sync getTigerGraphData<N extends InputNode, L extends InputLink> (vertex_type: Array<string>, edge_type: Array<string>) : Promise<{ nodes: N[]; links: L[]; }>
```

Reply:
@GenericP3rson Actually it'll be better to define generics on the class level (since we have a class now):
```ts
export class TigerGraphConnection<N extends InputNode, L extends InputLink> {
...
}
```